### PR TITLE
chore: refine types and onChange handlers

### PR DESCRIPTION
**Pull request by hein-j**
*Description of changes:*
- Change handlers and types to account for differences across component types.
Verified by creating a form. E2E test would be useful to prevent regressions here. That will follow in another PR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/forms/event-targets.ts [hunk 1]
```diff
@@ -17,57 +17,101 @@ import { DataFieldDataType } from '@aws-amplify/codegen-ui';
...
+  let defaultIdentifier: string | BindingName =
+    fieldTypeToExpressionMap[fieldType]?.identifier ?? expressionMap.destructuredValue;
   switch (dataType) {
-    // const value = Number(new Date(e.target.value));
     case 'AWSTimestamp':
+      // value = Number(new Date(e.target.value));
+      defaultIdentifier = expressionMap.value;
       expression = factory.createCallExpression(factory.createIdentifier('Number'), undefined, [
         factory.createNewExpression(factory.createIdentifier('Date'), undefined, [
           factory.createPropertyAccessExpression(expression, factory.createIdentifier('value')),
```
Comment (SwaySway) [inline packages/codegen-ui-react/lib/forms/event-targets.ts:81]: nit: we can re-use the value here as well from the expressionMap

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap [hunk 2]
```diff
@@ -1897,7 +1933,7 @@ export declare type InputGalleryCreateFormInputValues<useBase extends boolean =
     attend?: UseBaseOrValidationType<useBase, boolean>;
     maybeSlide?: UseBaseOrValidationType<useBase, boolean>;
     maybeCheck?: UseBaseOrValidationType<useBase, boolean>;
-    arrayTypeField?: UseBaseOrValidationType<useBase, string>;
+    arrayTypeField?: UseBaseOrValidationType<useBase, string[]>;
```
Comment (SwaySway) [inline packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap:1936]: i think we run validation on each element in the array as opposed to passing the whole list

File: packages/codegen-ui-react/lib/forms/event-targets.ts (hunk 1 above)
Comment (hein-j) [reply packages/codegen-ui-react/lib/forms/event-targets.ts]: nit you say?? this prevented a regression. thanks!

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap (hunk 2 above)
Comment (hein-j) [reply packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap]: Ok, I reverted the change to array types for now. If I find it needs changes, I'll PR that separately.

**Review by cwoolum — APPROVED** [review]
LGTM

**Review by letsbelopez — APPROVED** [review]

**Review by SwaySway — APPROVED** [review]
lgtm 🚀

**Pull request merged**